lib/compiler/passes/generate-js.js

```diff
@@ -788,15 +788,13 @@ function generateJS(ast, options) {
       "}",
       "",
       "function peg$SyntaxError(message, expected, found, location) {",
-      "  this.message = message;",
-      "  this.expected = expected;",
-      "  this.found = found;",
-      "  this.location = location;",
-      "  this.name = \"SyntaxError\";",
-      "",
-      "  if (typeof Error.captureStackTrace === \"function\") {",
-      "    Error.captureStackTrace(this, peg$SyntaxError);",
-      "  }",
+      "  const self = Error.call(this, message);",
+      "  Object.setPrototypeOf(self, peg$SyntaxError.prototype);",
```

Comment:
Right now generated parser is still a ES5 compatible code. As I remember, `const` available only from ES6, the same for `Object.setPrototypeOf`.

So this is a breaking change. I do not mind, because I myself plan to do some PRs that are likely to be breaking.

Just do not forget to update also a list of supported targets in the readme.

Reply:
const->var is an easy fix that was just carelessness on my part.  `Object.setPrototypeOf` is supported in IE11+, and I think we're still trying to support IE9 for the moment.  We can add a [polyfill](https://github.com/wesleytodd/setprototypeof/blob/master/index.js), or just accept slightly-incorrect behavior if  `Object.setPrototypeOf` doesn't exist.

Reply:
"slightly-incorrect" here looks like this:

```
Uncaught [Error [SyntaxError]: ... ]
```

instead of this:

```
[peg$SyntaxError: ... ]
```

which i think i can live with.

Reply:
So this polyfill [isn't actually possible](https://github.com/Financial-Times/polyfill-library/issues/47#issuecomment-609509177)

[I've made React 15 work in IE7](https://github.com/StoneCypher/preact_ie7)

This is a much uglier thing than you might expect

As a point, does it really matter?  If they weren't getting a nice class inheritance before, there's no sense establishing a backwards compatibility they never had in the first place

As long as it doesn't catch fire in new and unique ways, "compatability" can in their case be relegated to "eh, good enough" IMO.  If it was already broken, we don't have to fix it